### DISCUSSION TITLE
My app crashes on launch, error: "rejected due to the app being denylisted."

### DISCUSSION THREAD

**Discussion by BrandLibel** [airsdk/Adobe-Runtime-Support#881] [Q&A]
The full error I'm getting is `Snapshot generation request for bundleID: example.bundle.id rejected due to the app being denylisted.`

I found only 2 other results when searching for this error:

1. https://stackoverflow.com/questions/65203246/nativescript-app-crashing-on-ios-14-2-when-using-an-mdm
2. https://developer.apple.com/forums/thread/674421

There wasn't anything helpful in those results; some people said that the crashing was fixed after the iOS 14.4 update. I'm on 14.6 but still getting this error.

The error message is completely incomprehensible to me. What does "Snapshot generation" and "denylisted" mean?

My app is being installed Adhoc onto my iPad Air 2.

I'm packaging the app with this command: `adt -package -target ipa-ad-hoc -storetype pkcs12 -keystore cert/ios-distribution-05-16-2021.p12 -provisioning-profile cert/iosdistradhoc-05-16-2021.mobileprovision dist/MyApp.ipa MyApp-app.xml -extdir ane MyApp.swf icons
`

**Comment by ajwfrost** [comment]
Hi @BrandLibel  - are you able to check with a really simple/empty application that you build and deploy the same way, to see if this also faces the same issue? And do you know whether it was working with a different iOS version (can you test it on a different device/iOS version?)

In the two articles you mentioned
- the first, as you say, was an Apple bug that they fixed in 14.4
- the second shouldn't apply as we should be building as executables already, but may be worth a check...
Could you please try unzipping your ipa file and checking the file type? e.g.
```
mkdir unzipped && cd unzipped
unzip ../dist/MyApp.ipa
cd Payload/MyApp.app
otool -v -h MyApp
```
thanks

**Reply by BrandLibel** [reply]
Thank you. The `filetype` was `EXECUTE`. Also, when I tried with a blank / very simple application, deployed the same way, it worked.

It turned out the original error message I got was definitely related to the iOS Launch Screen.

My app manifest file had a value for a launch screen:

```
<iPhone>
   ...
  <key>UILaunchStoryboardName</key>
  <string>LaunchScreen</string> 
</iPhone>
```

Removing the <key> and <string> tags worked. I assume the problem was because I specified a launch screen in the manifest but did not actually provide the file in my application.
